docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Data.puml)
+
+The `ModuleList` class serves as the main data storage class for the program, and is always instantiated when the program is started. It holds:
+* A `Module` object representing the General Tasks list. This `Module` is instantiated upon `ModuleList`'s creation and is meant to be the "default" module for all uncategorised or miscellaneous tasks.
+* An `ArrayList` containing all user-created modules.
+
+The `Module` class serves as a wrapper around a `TaskList`, providing additional attributes including the module code and module description. Within the context of Mod Happy, modules can be viewed as task categories with names, descriptions and other attributes; for this reason, the General Tasks list is implemented as a `Module` under the hood.
+
+> 📔 <span style="color:#00bb00">**NOTE:**</span>
+>
+> An alternative method of implementing `ModuleList` is shown below, where the default General Tasks
+list is simply represented as a `TaskList` instead of a full-fledged `Module`.
+>
+> ![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/DataAlternative.puml)
+>
+> While this model is arguably closer to real life, the program logic would have to operate on different object types depending on whether a given `Task` belongs to a user-created Module or the default General Tasks list. This was deemed to increase coupling and introduce too much unnecessary clutter to the code, hence it was not used.
+
+<br>
+
+### 5.4. Command Component
+
+The `Command` component is in charge of actually executing the operations requested by the user.
+
+The following partial class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/CommandClassDiagram.puml)
+
+All commands inherit the abstract `Command` class and must contain an `execute()` method. The program logic that must be executed to fulfil the requested command is implemented in this method. Additionally, `execute()` returns any command output to be displayed to the user as feedback.
+
+`Command` instances are generated by their corresponding `Parser` classes (e.g. `AddCommand` is constructed by `AddParser`) and executed by `Main`.
+
+<br>
+
+### 5.5. Storage Component
+
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
+
+Storage is an interface supporting write/read data to/from computer storage:
```

Comment:
For readability purposes, perhaps it would be better to enclose all mentions of classes or methods in backticks i.e. (`Storage`), which is what you did in previous sections.